### PR TITLE
Advertise device IP and service status

### PR DESCRIPTION
**Pull request by ubnt-michal-wojcik**

### CODE REVIEW

File: lib/bleno.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ Bleno.prototype.onDisconnect = function(clientAddress) {
   this.emit('disconnect', clientAddress);
 };
 
-Bleno.prototype.startAdvertising = function(name, serviceUuids, mac, callback) {
+Bleno.prototype.startAdvertising = function(name, serviceUuids, mac, ip, serviceStatus, callback) {
```
Comment (ubnt-km) [inline lib/bleno.js]: if this can be called by caller outside of this module, it may cause problem.

File: lib/hci-socket/gap.js [hunk 2]
```diff
@@ -119,13 +115,73 @@ Gap.prototype.startAdvertising = function(name, serviceUuids, mac) {
     }
   }
 
+  // Scan Response Data
+  if (name && name.length) {
```
Comment (ubnt-km) [inline lib/hci-socket/gap.js]: not sure if `if(typeof name !== "undefined")` is better.

File: lib/hci-socket/gap.js [hunk 3]
```diff
@@ -59,12 +56,16 @@ Gap.prototype.startAdvertising = function(name, serviceUuids, mac) {
     advertisementDataLength += 2 + 16 * serviceUuids128bit.length;
   }
 
-  if (mac && mac.length === 12) {
-    advertisementDataLength += 10;
+  if (name && name.length) {
+    advertisementDataLength += 2 + name.length;
+  }
+
+  if (advertisementDataLength > 31) {
```
Comment (piotrcholewczuk) [inline lib/hci-socket/gap.js]: Please make constant for `31` as `MAX_PACKET_LENGTH`

File: lib/hci-socket/gap.js [hunk 4]
```diff
@@ -99,33 +100,58 @@ Gap.prototype.startAdvertising = function(name, serviceUuids, mac) {
     }
   }
 
-  if (mac && mac.length === 12) {
-    // Length = 0x09 = 1 byte (0x16 - service type) + 2 bytes service UUID (0x2A25 - serial number) + 6 bytes MAC
-    advertisementData.writeUInt8(0x09, advertisementDataOffset);
-    advertisementDataOffset++;
+  // Shortened name requires at least 3 bytes left in Advertisement Packet
+  if (name && name.length && advertisementDataOffset < 29) {
```
Comment (piotrcholewczuk) [inline lib/hci-socket/gap.js]: Please use `MAX_PACKET_LENGTH - 2`

File: lib/hci-socket/gap.js [hunk 5]
```diff
@@ -99,33 +100,58 @@ Gap.prototype.startAdvertising = function(name, serviceUuids, mac) {
...
+      // Write 2B Service Data UUID
+      for (j in serviceData[i].uuid) {
+        scanData.writeUInt8(serviceData[i].uuid[j], scanDataOffset);
+        scanDataOffset++;
+      }
+      // Write Service Data
+      for (j in serviceData[i].data) {
+        scanData.writeUInt8(serviceData[i].data[j], scanDataOffset);
+        scanDataOffset++;
+      }
```
Comment (piotrcholewczuk) [inline lib/hci-socket/gap.js]: I'm not convinced to keep data as array - please make a note about that format.

**Pull request merged**